tools/autograd/load_derivatives.py

```diff
@@ -85,11 +87,147 @@ def create_derivative(f: NativeFunction, formula: str, var_names: Tuple[str, ...
+            if not len(derivatives) == 1:
+                raise RuntimeError(f"Derivative definition of {defn_name} in derivatives.yaml defines the "
+                                   "forward definition of gradient as element_wise but it does not "
+                                   "defines the gradient formula for its argument which is required.")
+
+            backward_formula = derivatives[0].original_formula
+            input_name = args_with_derivatives[0].name
+
+            def repl(m: Any) -> str:
+                return f"{m.group(1)}{input_name}_t{m.group(2)}"
+            fw_formula = re.sub(IDENT_REGEX.format("grad"), repl, backward_formula)
+
+            for arg in args_with_derivatives:
+                arg_name = arg.name
+
+                def repl(m: Any) -> str:
+                    return f"{m.group(1)}{arg_name}_p{m.group(2)}"
+                fw_formula = re.sub(IDENT_REGEX.format(arg_name), repl, fw_formula)
+
+            required_inputs_tangent = tuple(all_arg_names)
+            formula = fw_formula
+        elif formula == "auto_linear":
+            if len(forward_derivatives) > 1:
+                raise RuntimeError(f"Derivative definition of {defn_name} in derivatives.yaml defines the "
+                                   "forward definition of gradient as linear but this only works "
+                                   "for functions with a single differentiable output.")
+
+            diff_arg_names = [arg.name for arg in args_with_derivatives]
+            assert len(diff_arg_names) > 0
+
+            new_args = []
+            for arg_name in all_arg_names:
+                if arg_name in diff_arg_names:
+                    arg_name = arg_name + "_t"
+                new_args.append(arg_name)
+
+            if Variant.function in f.variants:
+                fw_formula = "at::{}({})".format(defn_name, ", ".join(new_args))
+            else:
+                assert f.func.kind() is not SchemaKind.inplace
+                assert Variant.method in f.variants
+                fw_formula = "{}.{}({})".format(new_args[0], defn_name, ", ".join(new_args[1:]))
+
+            required_inputs_tangent = tuple(diff_arg_names)
+            formula = fw_formula
+
+        # During forward formula, we use the primal instead of the input Tensors
+        required_inputs_primal = find_required_inputs(formula, "_p")
+
+        updated_derivatives.append(ForwardDerivative(
+            formula=formula,
+            var_name=defn.var_name,
+            var_type=defn.var_type,
+            required_inputs_fw_grad=required_inputs_tangent,
+            required_inputs_primal=required_inputs_primal))
+
+    return updated_derivatives
+
+def is_forward_derivative_definition(all_arg_names: List[str], names: Tuple[str, ...]) -> bool:
+    if len(names) > 1:
+        # Forward definition are always for a single output at a time
+        return False
+    name = names[0]
+    if name not in all_arg_names:
+        return True
```

Comment:
Wouldn't it be more direct to test if this is one of the distinguished output names (`result`) or if it is one of the output names?

Reply:
(or, alternately, "is_backward_derivative_definition") here

Reply:
The problem is that multi-output functions with default output name end up with named `resultX` with X being the index of the output. It makes it harder to directly see if it is an output (especially because nothing prevents the user from naming an arguments result2 as long as they specify the name of the output).